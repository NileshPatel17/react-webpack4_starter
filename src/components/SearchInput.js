import React from 'react';
import cxs from 'cxs';
import PropTypes from 'prop-types';

const SearchInput = ({ searchPredicate, handleSearchChange }) => (
  <label>
    <span
      className={cxs({
        position: 'absolute !important',
        clip: 'rect(1px, 1px, 1px, 1px)'
      })}
    >
      Filter Search
    </span>
    <input
      placeholder="Filter by name"
      type="text"
      autoFocus
      className={cxs({
        padding: '1rem',
        width: '100%',
        borderRadius: '2px',
        border: '1px solid #adb5bd',
        color: '#000',
        overflow: 'visible',
        margin: 0,
        boxSizing: 'border-box'
      })}
      value={searchPredicate}
      onChange={e => {
        handleSearchChange(e.target.value);
      }}
    />
  </label>
);
SearchInput.propTypes = {
  searchPredicate: PropTypes.string.isRequired,
  handleSearchChange: PropTypes.func.isRequired
};
export default SearchInput;
