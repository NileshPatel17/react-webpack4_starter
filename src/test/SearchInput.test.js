import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { assertPropTypes } from 'check-prop-types';
import PropTypes from 'prop-types';
import SearchInput from '../components/SearchInput';

describe('<SearchInput />', () => {
  const inputText = 'refs';
  const mockSearchChange = jest.fn();
  const props = { searchPredicate: inputText, handleSearchChange: mockSearchChange };
  const component = shallow(<SearchInput {...props} />);

  it('SearchInput component should render as expected', () => {
    const tree = toJson(component);
    expect(component).toMatchSnapshot();
  });

  it('has "searchPredicate" a required SearchInput propTypes', () => {
    assertPropTypes(
      SearchInput.propTypes,
      { searchPredicate: 'refs', handleSearchChange: () => {} },
      SearchInput.searchPredicate,
      SearchInput.searchPredicate
    );
  });

  it('has "handleSearchChange" a required function propTypes', () => {
    expect(SearchInput.propTypes.handleSearchChange).toBe(PropTypes.func.isRequired);
  });

  describe('When the user types into the input', () => {
    // beforeEach(() => {
    //   component.find('input').simulate('change', { target: { value: inputText } });
    // });
    // it('should call mock function when use types into the input', () => {
    //   component.find('input').simulate('change', { target: { value: inputText } });
    //   expect(typeof component.find('input').node.props.value).toBe(inputText);
    //   expect(mockSearchChange).toHaveBeenCalled();
    // });
  });
});
