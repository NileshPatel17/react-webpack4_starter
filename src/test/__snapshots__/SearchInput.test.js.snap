// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> SearchInput component should render as expected 1`] = `
<label>
  <span
    className="x0 x1"
  >
    Filter Search:
  </span>
  <input
    autoFocus={true}
    className="x2 x3 x4 x5 x6 x7 x8 x9"
    onChange={[Function]}
    placeholder="Filter by name"
    type="text"
    value="refs"
  />
</label>
`;
